cmd/lncli/cmd_pay.go

```diff
@@ -467,20 +454,113 @@ func trackPayment(ctx *cli.Context) error {
 		return err
 	}
 
+	_, err = printLivePayment(stream)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// printLivePayment uses advanced terminal control codes to rewrite the output
+// with the latest payment state. This continues until the payment reaches a
+// final state.
+func printLivePayment(stream routerrpc.Router_TrackPaymentClient) (
+	*lnrpc.Payment, error) {
+
+	writer := uilive.New()
+	writer.Start()
+	defer writer.Stop()
 	for {
-		status, err := stream.Recv()
+		payment, err := stream.Recv()
 		if err != nil {
-			return err
+			return nil, err
 		}
 
-		printRespJSON(status)
+		table := formatPayment(payment)
```

Comment:
I'd add an option to select format type, as some users may find it hard to parse the table (eg. on small terminals), or may want to parse it with `jq` or other tooling.

Reply:
Yes, the only thing is that the output wasn't parseable by `jq`. It was a concatenation of multiple json messages.

Reply:
I don't know what to do with this. I think a basic question is whether `lncli` is meant to be used in automated processes or as a user interface. Given the large number of options to programmatically interface with `lnd`, I tend to see it as a user interface. The json output is more like an easy way to get the data on the screen.

Reply:
For sure the table output is the nicest for human consumption. Maybe even easy to process from bash/python script if someone feels like it. Maybe an other option could be a log like output? Something where state changes are easy to debug with grep. wdyt?

Reply:
Not sure if it is easy to parse the formatted, rewritten table.

This PR makes the output readable for humans which I think is enough. But it comes down to whether we want to support automated processes / bash / python / grep. We also don't deprecate things in `lncli`, so from a maintenance perspective it isn't ideal either to use it for automation.

Reply:
perhaps just a `-r` would suffice for being able to toggle back to the raw json output? it's true we don't recommend people building production things against lncli, it's primarily meant as a debugging tool for hoomans

imo it's nice to finally see lncli moving away from just logging json responses by default, and trying to present the data in a more easily consumable format.

Reply:
I can add `-r`, but do we still want that for the intermediate state updates? That was a concatenation of multiple json messages which isn't parseable as a whole. Maybe best to just do the final update in that case?

Reply:
It seems there are two variables here.
1. Whether or not to send non-terminal events, say `--block`.
2. Whether or not to format as table or json, say `-r`.

Both I think should be lncli flags, but only `--block` needs to be an rpc flag.

Turns out this also provides REST clients that don't support server-side streaming via keep-alive connections to use the streaming endpoint in the typical request/response patern. The url just needs to contain `?block=true` and only the terminal response is sent back before the connection is closed.

Reply:
Thanks for the reminder about how to deal with non-streaming rest. Added the `NoInflightUpdates` flag. I think it is more descriptive than `Block`. This option already existed in `lncli` as `--show_inflight`. Didn't change that. 

Added the option to output raw json with `--json``.

Reply:
> This option already existed in lncli as --show_inflight. Didn't change that.

Wasn't this option added in this release? If wanted to change it now is the time.

Reply:
Yes, I can rename it. But to what? I agree that `show_inflight` isn't ideal. We could mirror the rpc field, `--inflight_updates`. I prefer that over `--block`.

Reply:
Renamed to `--inflight_updates`, in line with the rpc field. But always showing inflight updates for the table view